mmv1/third_party/terraform/resources/resource_compute_security_policy.go.erb

```diff
@@ -324,17 +324,35 @@ func resourceComputeSecurityPolicy() *schema.Resource {
 					Schema: map[string]*schema.Schema{
 						"json_parsing": {
 							Type: schema.TypeString,
-								Optional: true,
-								Computed: true,
-								ValidateFunc: validation.StringInSlice([]string{"DISABLED", "STANDARD"}, false),
-								Description: `JSON body parsing. Supported values include: "DISABLED", "STANDARD".`,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{"DISABLED", "STANDARD"}, false),
+							Description: `JSON body parsing. Supported values include: "DISABLED", "STANDARD".`,
+						},
+						"json_custom_config": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
+							Description: `Custom configuration to apply the JSON parsing. Only applicable when JSON parsing is set to STANDARD.`,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"content_types": {
+										Type: schema.TypeSet,
+										Required: true,
+										MinItems: 1,
```

Comment:
My `TypeSet` memory could be fuzzy, but would this field be able to set to 0 items without this? Would the API accept it?

Reply:
This actually took me down a bit of a rabbit hole. To answer your questions, yes you could set this to `[]` if we removed the `MinItems: 1`, and the API does accept this, but it introduces some weird behaviors related to the way Terraform handles zero-values. Basically, `content_types: []` gets treated as if `content_types` is not set at all. And since this is the only field in the `json_custom_config` block, the block is considered empty/unset as well.

I'm torn on the best way to handle this though, because I think you bring up a really important use-case, which is the ability to clear out `content_types` after they have been set. I think we need to remove the `MinItems: 1`, but @c2thorn do you know of an existing pattern for handling these zero-values? The only solution I've come up with that seems to work is to change the `expand*` function to explicitly set an empty `JsonCustomConfig` when we hit this case. That seems to ensure the field is included in requests, and therefore subsequent responses as well.

```
if configured[0] == nil {
	return &compute.SecurityPolicyAdvancedOptionsConfigJsonCustomConfig{}
}
```

Reply:
I think I missed that this was the only field within `json_custom_config` in my first passthrough. Given the `json_custom_config` is optional+computed, the only way to explicitly remove it is to set the entire `json_custom_config` to an empty list. This is gets the job done for `content_types` as well, so I think we are safe to leave `MinItems: 1`

Reply:
Hmm, I'm not sure if this solves the root issue with zero-values, but I could be missing something. If we keep `MinItems: 1`, I'm assuming that we would also need to make `content_types` optional+computed instead of required, and then expect the user to clear the list with the following Terraform:

```
resource "google_compute_security_policy" "policy" {
  advanced_options_config {
    json_custom_config {}
  }
}
```

Is that what you were suggesting?

I have not been able to get a solution like this working, presumably because the `json_custom_config {}` is a zero-value that Terraform ignores, so it doesn't get sent to the server at all. The result is that the `content_types` cannot be cleared. I also tried using `ForceSendFields`, but it did not seem to help.

Reply:
Apologies for the back and forth, we are getting into specific behavior that has been a while since I've done myself. I've dug into this more, and I think I have a solid idea of what's going on, but with the caveat that I still have not actually tried it out.

I believe the empty `json_custom_config` doesn't send because of https://github.com/GoogleCloudPlatform/magic-modules/pull/6521/files#diff-7f40a6d1420bc5b6fdfefde5279bb9ff7d5088b470f4133930e89b26c4110377R840 where we nil it out. I realize that this simply follows the pattern set in this resource, like in the parent `advanced_options_config` which is also optional+computed.
Looking to a popular resource we have, I've found another example of a O+C list where we do something different: https://github.com/GoogleCloudPlatform/magic-modules/blob/add-security-policy-json-custom-config/mmv1/third_party/terraform/resources/resource_container_cluster.go.erb#L3809
If the list is size 0 or nil, we still send the object. I _believe_ this is what we would have to do here as well.

Putting all of this into historical context, container cluster probably does this differently because we've been specifically asked to allow this to happen. Given that `advanced_options_config` probably also cannot be zero'd out, we are likely pretty inconsistent with this across the provider. My personal recommendation would be to try out sending a non-nil object, but if you still run into difficulties, handling this edge case is more trouble than it is worth right now.

Reply:
Thanks for the thorough response and context! I'll give this a try and report back.

Reply:
So I tried a few additional things here, but in short, I couldn't get the empty `json_custom_config` to send the way we were hoping to clear `content_types`. I believe Terraform cannot tell between this config and an intention to ignore management of the field/subfields.

Instead, I ended up circling back to the theory in my initial comment, but tweaking it to match the `container_cluster` example you shared: https://github.com/GoogleCloudPlatform/magic-modules/pull/6521/commits/7aa6e5bc906fed55c2079d4ed5a95542a315bd73

The end result is that `content_types` can be cleared with config like this:
```
resource "google_compute_security_policy" "policy" {
  advanced_options_config {
    json_custom_config {
      content_types: []
    }
  }
}
```
and if `json_custom_config` is not set at all, Terraform will not try to manage it (I think this is important for existing users).

From what I could tell in my testing, this implementation covers everything we need, but let me know if something still seems off.

Reply:
Thanks for diving deep into this initially minor question. This will be helpful for users who fall into this edge case.